nodejs/week3/practices/exercise-template/src/backend/api/concerts.js

```diff
@@ -0,0 +1,169 @@
+            const requestPrice=Number(request.query.maxPrice);
+            const priceFinder = concerts.some(concert=> Number(concert.price) < requestPrice);
+            const requestedConcert = await concerts.filter(concert=> Number(concert.price)<requestPrice);
+          
+    
+            if(priceFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+ 
+          }else if(request.query.hasOwnProperty("title")){
+           
+           
+            const requestTitle=request.query.title.toLowerCase();
+            const titleFinder = concerts.some(concert=> concert.title.toLowerCase() ===requestTitle);
+            const requestedConcert = await concerts.filter(concert=>concert.title.toLowerCase() ===requestTitle);
+           
+    
+            if(titleFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+           
+          }else if(request.query.hasOwnProperty("createdAfter")){
+            
+            const requestDate=new Date(request.query.createdAfter);
+            const concertFinder = concerts.some(concert=> new Date(concert.created_date) > requestDate);
+            const requestedConcert = await concerts.filter(concert => new Date(concert.created_date) > requestDate);
+           
+    
+            if(concertFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+   
+          }else if(request.query.hasOwnProperty("band")){
+            const requestBand=request.query.band.toLowerCase();
+            const concertFinder = concerts.some(concert=> concert.band.toLowerCase() ===requestBand);
+            const requestedConcert = await concerts.filter(concert=>concert.band.toLowerCase() ===requestBand);
+           
+    
+            if(concertFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+          }else{
+            response.json(concerts);
+          }
+ 
+  } catch (error) {
+    throw error;
+  }
+});
+  
+
+router.post("/", async (request, response) => {
+
+  try {
+     // knex syntax for selecting things. Look up the documentation for knex for further info
+    const concerts = await knex('concerts').insert(request.body);
+   
+    // Alternatively could also insert in the way commented below
+    // but I personally prefer request.body method and using postman.
+   
+    // const concerts = await knex('concerts').insert( 
+    //   {"title": "spring festival",
+    // "band": "many singers",
+    // "venue": "spring celebration",
+    // "created_date": '2021-02-28 15:00:00',
+    // "performance_date": '2021-04-22 17:00:00',
```

Comment:
Always remember to clean up unwanted code before you commit and push your code.
and alternatively you can insert like in this comment but that'll be you hardcoding the values so inserting with the ``req.body`` is very right and what you should be doing. 💯